### PR TITLE
Warnings etc

### PR DESCRIPTION
**Pull request by dsvensson**
* Makes the debug variant of the cmake build work by fixing warnings.
* Fixes wasm build.
* Adds GitHub Action to protect against future breakage.

### CODE REVIEW

**Comment by dsvensson** [comment]
I'm very much not sure about the ODE commit. Don't know how to test it, but as it's a vector, I suspected that it should use quaternion rotation, as the rotation function expected a matrix.

Not sure if GitHub Actions is wanted or not, added to safeguard against future warnings. I only exposed the wasm artifacts as I expect those consumers are least likely to build themselves.

File: engine/gl/gl_heightmap.c [hunk 1]
```diff
@@ -5738,7 +5738,7 @@ void Terr_Brush_Draw(heightmap_t *hm, batch_t **batches, entity_t *e)
 		{
 			if (br->faces[j].relight && dorelight)
 			{
-				lightstyleindex_t styles[4] = {0,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE};
+				lightstyleindex_t styles[MAXCPULIGHTMAPS] = {0,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE};
```
Comment (dsvensson) [inline engine/gl/gl_heightmap.c:5741]: ops.. perhaps should match size, or perhaps drop the last two ones there as it breaks iteration on first invalid.

**Comment by dsvensson** [comment]
(for context, review in discord)

**Comment by dsvensson** [comment]
Dropped ODE commit as I don't really know how to test it. Also dropped FFMPEG commit as baseline code for avplug does not work with my FFMPEG, nor does fixing the warnings, so perhaps some other API change in play.

File: engine/Makefile [hunk 2]
```diff
@@ -1726,6 +1726,7 @@ ifeq ($(FTE_TARGET),web)
 	EMCC_CFLAGS= -DFTE_TARGET_WEB
 	EMCC_LDFLAGS=$(EMCC_CFLAGS) $(JSLIBS) $(WEB_PREJS)
 #	EMCC_CFLAGS+= -s BINARYEN_TRAP_MODE='clamp'		#fix bigfloat->int rounding crashes
+	EMCC_CFLAGS+= -s USE_ZLIB=1
```
Comment (Shpoike) [inline engine/Makefile:1729]: This should not be needed if you're using `make makelibs FTE_TARGET=web` before building.

Comment (dsvensson) [reply engine/Makefile:1729]: Thanks for reminding me, humorously enough I'm currently flashing memtest86 to a USB-pin 🙃

**Pull request merged**